docs/guides/multisite_custom.rst

```diff
@@ -0,0 +1,230 @@
+
+        expecting a root page to exist that has the appropriate path
+
+        otherwise, the provided uri path will be used and the system will
+        try to retrieve a probably existing page via the ORM.
+        """
+        actual_site = current_site()
+        if not path:
+            # make sure that the user is redirected to the correct url
+            # we expect a root page with custom path /<default_language>/ to exist here
+            redirect_path = "/{}/".format(actual_site.default_language)
+            return redirect(redirect_path)
+
+        actual_path = "/{}/".format(path)
+        page = get_object_or_404(
+            Page.objects.active(),
+            path=actual_path
+        )
+        page.activate_language(request)
+        ancestors = page.ancestors().reverse()
+        meta_data_providers = [page] + list(ancestors) + [page.site]
+        return render(
+            request,
+            page.type.template_name,
+            {
+                "page": page,
+                "regions": Regions.from_item(
+                    page,
+                    renderer=renderer,
+                    inherit_from=ancestors,
+                    timeout=60
+                ),
+                "meta_tags": meta_tags(
+                    meta_data_providers,
+                    request=request,
+                    site_name=page.site.name
+                )
+            },
+        )
+
+Make sure that your custom site model get registered with the Django ORM
+before your page model gets registered, otherwise there will be an exception
+telling you that the site model configured in ``FEINCMS3_SITES_SITE_MODEL``
+does not exist.
+
+Remember to update your migrations as well:
+
+.. code-block:: shell
+
+    ./manage.py makemigrations
+    ./manage.py migrate
+
+
+Implementing a Custom Site Model Admin
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: python
+
+    from django import forms
+    from django.conf import settings
+    from django.contrib import admin
+    from django.utils.translation import gettext_lazy as _
+
+    from feincms3_meta.models import MetaMixin
+
+    from .models import Site
+
+
+    class SiteForm(forms.ModelForm):
+        default_language = Site._meta.get_field("default_language").formfield(
+            choices=[("", "----------")] + list(settings.LANGUAGES),
+        )
+
+        class Meta:
+            model = Site
+            fields = "__all__"
+
+
+    class SiteDefaultLanguageListFilter(admin.SimpleListFilter):
+
+        title = _("default language")
+        parameter_name = "default_language"
+
+        def lookups(self, request, model_admin):
+            return settings.LANGUAGES
+
+        def queryset(self, request, queryset):
+            if self.value():
+                return queryset.filter(default_language=self.value())
+            else:
+                return queryset.all()
+
+
+    @admin.register(Site)
+    class SiteAdmin(admin.ModelAdmin):
```

Comment:
Again, is there a reason why you couldn't base this off `feincms3_sites.admin.SiteAdmin`?

The default language filter could also be added to `feincms3_sites`' core so that you wouldn't have to copy this code into individual projects if that was possible.